pydantic_ai_slim/pydantic_ai/_output.py

```diff
@@ -113,128 +118,460 @@ def __init__(self, tool_retry: _messages.RetryPromptPart):
 class ToolOutput(Generic[OutputDataT]):
     """Marker class to use tools for outputs, and customize the tool."""
 
-    output_type: SimpleOutputType[OutputDataT]
+    output: OutputTypeOrFunction[OutputDataT]
     name: str | None
     description: str | None
     max_retries: int | None
     strict: bool | None
 
     def __init__(
         self,
-        type_: SimpleOutputType[OutputDataT],
+        type_: OutputTypeOrFunction[OutputDataT],
         *,
         name: str | None = None,
         description: str | None = None,
         max_retries: int | None = None,
         strict: bool | None = None,
     ):
-        self.output_type = type_
+        self.output = type_
         self.name = name
         self.description = description
         self.max_retries = max_retries
         self.strict = strict
 
 
+@dataclass
+class TextOutput(Generic[OutputDataT]):
+    """Marker class to use text output with an output function."""
+
+    output_function: TextOutputFunction[OutputDataT]
+
+
+@dataclass(init=False)
+class StructuredTextOutput(Generic[OutputDataT]):
```

Comment:
My preference is to have two markers `ModelStructuredOutput` and `PromptedStructuredOutput`.